eclipse-scout-core/src/util/objects.ts

```diff
@@ -741,6 +741,13 @@ export const objects = {
     return Object.keys(obj).length === 0;
   },
 
+  /**
+   * @returns true if the first parameter is the same or a subclass of the second parameter
+   */
+  isSameOrExtendsClass<TClass>(class1: any, class2: TClass): boolean {
```

Comment:
improve flow control by

```suggestion
  isSameOrExtendsClass<TSuperClass>(subClass: any, superClass: TSuperClass): subClass is TSuperClass {
```